File: backend/models/Complaint.js
const mongoose = require('mongoose');

const complaintSchema = new mongoose.Schema({
  studentId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  studentName: String,
  busId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Bus'
  },
  type: {
    type: String,
    required: true,
    enum: ['Delay', 'Behavior', 'Maintenance', 'Other']
  },
  description: {
    type: String,
    required: true
  },
  status: {
    type: String,
    enum: ['pending', 'in-progress', 'resolved'],
    default: 'pending'
  },
  remarks: {
    type: String,
    default: ''
  }
}, { timestamps: true });

module.exports = mongoose.model('Complaint', complaintSchema);

File: android-app/app/src/main/java/com/college/bustrack/models/Complaint.java
package com.college.bustrack.models;

import com.google.gson.annotations.SerializedName;

public class Complaint {
    @SerializedName("_id")
    private String id;
    private String studentId;
    private String studentName;
    private String busId;
    private String type; // e.g., "Delay", "Behavior", "Maintenance"
    private String description;
    private String status; // "pending", "in-progress", "resolved"
    private String remarks;
    private String createdAt;

    // Getters and Setters
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    public String getStudentId() { return studentId; }
    public void setStudentId(String studentId) { this.studentId = studentId; }
    public String getStudentName() { return studentName; }
    public void setStudentName(String studentName) { this.studentName = studentName; }
    public String getBusId() { return busId; }
    public void setBusId(String busId) { this.busId = busId; }
    public String getType() { return type; }
    public void setType(String type) { this.type = type; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
    public String getRemarks() { return remarks; }
    public void setRemarks(String remarks) { this.remarks = remarks; }
    public String getCreatedAt() { return createdAt; }
    public void setCreatedAt(String createdAt) { this.createdAt = createdAt; }
}
